hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -171,9 +172,17 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
             false /* throw NoSuchObjectException if the table doesn't exist */);
         return null;
       });
-
       if (purge && lastMetadata != null) {
         CatalogUtil.dropTableData(ops.io(), lastMetadata);
```

Comment:
If we own the full table dir - can we expect that all of the files are in the given dir?
We might not need this `CatalogUtil.dropTableData(ops.io(), lastMetadata)` call when `shouldDropFullTableFolder` is `true`

Reply:
I think we can't expect that all of the files related to a table are in it's own directory. At least it's not enforced after setting the new tableproperty to keep all the files in metadata.location(). Additionally this tableproperty could be set and unset without checking the locations of the existing files.